Resources/MessageQueue/StompMQConnector.py

```diff
@@ -46,53 +58,55 @@ def setupConnection(self, parameters=None):
+
     host = self.parameters.get('Host')
-    port = self.parameters.get('Port', 61613)
+    port = self.parameters.get('Port', StompMQConnector.PORT)
     vhost = self.parameters.get('VHost')
 
     sslVersion = self.parameters.get('SSLVersion')
     hostcert = self.parameters.get('HostCertificate')
     hostkey = self.parameters.get('HostKey')
-    # get local key and certificate if not available via configuration
-    if sslVersion and not (hostcert or hostkey):
-      paths = Locations.getHostCertificateAndKeyLocation()
-      if not paths:
-        return S_ERROR('Could not find a certificate!')
-      else:
-        hostcert = paths[0]
-        hostkey = paths[1]
-
-    try:
 
-      # get IP addresses of brokers
-      brokers = socket.gethostbyname_ex(host)
-      self.log.info('Broker name resolves to %s IP(s)' % len(brokers[2]))
+    connectionArgs = {'vhost': vhost,
+                      'keepalive': True,
+                      'reconnect_sleep_initial': reconnectSleepInitial,
+                      'reconnect_sleep_increase': reconnectSleepIncrease,
+                      'reconnect_sleep_max': reconnectSleepMax,
+                      'reconnect_sleep_jitter': reconnectSleepJitter,
+                      'reconnect_attempts_max': reconnectAttemptsMax}
 
-      if sslVersion is None:
-        pass
-      elif sslVersion == 'TLSv1':
+    if sslVersion is not None:
+      if sslVersion == 'TLSv1':
         sslVersion = ssl.PROTOCOL_TLSv1
+        # get local key and certificate if not available via configuration
+        if not (hostcert or hostkey):
+          paths = Locations.getHostCertificateAndKeyLocation()
+          if not paths:
+            return S_ERROR('Could not find a certificate!')
+          hostcert = paths[0]
+          hostkey = paths[1]
+        connectionArgs.update({
+                              'use_ssl': True,
+                              'ssl_version': sslVersion,
+                              'ssl_key_file': hostkey,
+                              'ssl_cert_file': hostcert})
       else:
         return S_ERROR(EMQCONN, 'Invalid SSL version provided: %s' % sslVersion)
 
-      for ip in brokers[2]:
-        if sslVersion:
-          self.connections[ip] = stomp.Connection(
-              [(ip, int(port))],
-              use_ssl=True,
-              ssl_version=sslVersion,
-              ssl_key_file=hostkey,
-              ssl_cert_file=hostcert,
-              vhost=vhost,
-              keepalive=True
-          )
-        else:
-          self.connections[ip] = stomp.Connection(
-              [(ip, int(port))],
-              vhost=vhost,
-              keepalive=True
-          )
+    try:
+      # Get IP addresses of brokers and ignoring two first returned arguments which are hostname and aliaslist.
+      _, _, ip_addresses = socket.gethostbyname_ex(host)
+      self.log.info('Broker name resolved,  to %s IP(s)' % len(ip_addresses))
```

Comment:
I meant to do:

```suggestion
      self.log.info('Broker name resolved',  'to %s IP(s)' % len(ip_addresses))
```

DIRAC loggers have a fixed part and a variable part (https://dirac.readthedocs.io/en/latest/DeveloperGuide/AddingNewComponents/Utilities/gLogger/gLogger/Basics/index.html#log-record-creation)
The fixed part is what can be used "for statistics" and the variable part carries what's specific to the log message itself.

Reply:
Thank you for the explanations. I changed it